dbt_coverage/__init__.py

```diff
@@ -109,48 +90,39 @@ class Catalog:
 
     tables: Dict[str, Table]
 
-    def filter_catalog(self, model_path_filter: List[str]) -> Catalog:
+    def filter_tables(self, model_path_filter: List[str]) -> Catalog:
         """
-        Filter ``Catalog``'s ``tables`` attribute to ``Tables`` that have the``model_path_filter``
-        value at the start of their ``original_file_path``.
+        Filters ``Catalog``'s ``tables`` attribute to ``Tables`` that have the
+        ``model_path_filter`` value at the start of their ``original_file_path``.
 
-        :param model_path_filter: the model_path string(s) to filter tables on, (matches using
-        the ``startswith`` operator)
+        Args:
+            model_path_filter: the model_path string(s) to filter tables on, (matches using
+                the ``startswith`` operator)
 
-        :returns: Catalog
-        :raises ValueError: if no ``Table`` in the ``tables`` Catalog attribute have an
-        ``original_file_path`` that contains any ``model_path_filter`` value
+        Returns:
+            New ``Catalog`` instance containing only ``Table``s that passed the filter
         """
-        filtered_tables = {}
 
-        original_tables_dict = {key: val for key, val in self.tables.items()}
-        for key, table in original_tables_dict.items():
-            for path in model_path_filter:
-                if table.original_file_path.startswith(path):
-                    filtered_tables[key] = table
-                    break
+        model_path_filter = tuple(model_path_filter)
+        tables = {
+            t_id: t
+            for t_id, t in self.tables.items()
+            if t.original_file_path.startswith(model_path_filter)
+        }
 
-        if len(filtered_tables) < 1:
-            logging.error("len(filtered_tables) < 1", exc_info=True)
-            raise ValueError(
```

Comment:
Moving the check to [`do_compute`](https://github.com/slidoapp/dbt-coverage/pull/57/files#diff-b6f93a17b34d0a435a739418b23a17c9caa347b9a88c183843f6b10b8c548a5bR819). We want to raise the exception about a CLI argument close to the point where we parse the arguments.